From ac607e1ee292968aa0e2200c0e8c8be5441240e4 Mon Sep 17 00:00:00 2001
From: Andreas Schwab <schwab@suse.de>
Date: Sat, 22 Nov 2025 11:29:43 +0100
Subject: [PATCH 2/2] gprofng: protect against standard library macros

The CALL_UTIL macro can expand to an unparsable expression of the argument
is a macro, like with the new const-preserving standard library macros in
C23.

	* gprofng/src/collector_module.h (CALL_UTIL): Add parens to not
	expand its argument if it is a function-like macro.

Upstream-Status: Backport
---
 gprofng/src/collector_module.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gprofng/src/collector_module.h b/gprofng/src/collector_module.h
index 81f50771975..03cc624255d 100644
--- a/gprofng/src/collector_module.h
+++ b/gprofng/src/collector_module.h
@@ -118,7 +118,7 @@ typedef struct CollectorUtilFuncs
 extern CollectorUtilFuncs __collector_util_funcs;
 extern int __collector_dlsym_guard;
 
-#define CALL_UTIL(x) __collector_util_funcs.x
+#define CALL_UTIL(x) (__collector_util_funcs.x)
 
 /* The following constants define the meaning of the "void *arg"
  * argument of getFrameInfo().
-- 
2.55.0

